homework/index.js

```diff
@@ -30,18 +30,119 @@
     return elem;
   }
 
+  // create reusable function (removeChildren),
+  // to remove children of an html element which is (the parent).
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+
+  // defined root outside the local scope of the function to be available to all functions.
+  const root = document.getElementById('root');
```

Comment:
It is customary to place global variables at the top of the file (in this case right after the opening curly brace, just before the `fetchJSON` function), so that they are the first thing you see when you open up the file. I'm not fond of global variables, as it is not immediately clear where they are used. Your comment _"defined root outside the local scope of the function to be available to all functions"_ describes exactly the problem: it can be used all over the place.

In this particular case, you also run the risk of the variable being undefined if someone decides to move your `app.js` file from end of the `body` tag to the `head` tag. In that case, when this statement runs your root `div` will not exist yet (that's why we use `window.onload` here).

If you place this line inside your `main` function and pass it as an argument to any function that needs it, you don't run this risk. As far as I can tell you only reference this `root` variable in your function `main` and in your function `createContributor`. The latter is only called by your function `main` so you could easily pass it as a parameter and get rid of the global variable:

```js
unction main(url) {
  const root = document.getElementById('root');

  fetchJSON(url, (err, data) => {
   ...

   createContributor(selectedRepo, repoContainer, root);
   ...
}
```

Reply:
U are right! Thanks a lot for your explanation. It's done!